### PR TITLE
Clean out headers and unify type aliasing

### PR DESCRIPTION
**Pull request by ZiyanShi**
### What problem does this PR solve?

Clean code base, improve readability.

What's Changed:

Clean out unused headers
Move aliasing visible to users to include/kvdk/types.hpp.
Move  internally used types to engine/aliasing.hpp and all internal headers include this file.

### Check List 

Tests 

- Unit test

Side effects

None

### CODE REVIEW

**Review by JiayuZzz — APPROVED** [review]

File: include/kvdk/types.hpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using CollectionIDType = std::uint64_t;
+using IndexType = std::int64_t;
+
+using UnixTimeType = std::int64_t;
+using ExpireTimeType = UnixTimeType;
+using TTLType = std::int64_t;
+
+/// TODO: helper functions to convert between ExpireTime and TTL
+constexpr ExpireTimeType kPersistTime = INT64_MAX;
+constexpr ExpireTimeType kInvalidExpireTime = -2;
```
Comment (sherry-1001) [inline include/kvdk/types.hpp:21]: is 0.

File: engine/collection.hpp [hunk 2]
```diff
@@ -10,13 +10,11 @@
...
 
 #include "alias.hpp"
-#include "kvdk/namespace.hpp"
 #include "kvdk/status.hpp"
 
 namespace KVDK_NAMESPACE {
 /// TODO: (ziyan) add expire_time field to Collection.
-
-// A collection of key-value pairs
+/// TODO: (ziyan) move collection.hpp to engine/
```
[the remark this answers is not in the document]
Comment (ZiyanShi) [reply engine/collection.hpp:17]: removed outdated comment

File: include/kvdk/types.hpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
+#pragma once
+#ifndef KVDKDEF_HPP
```
[the remark this answers is not in the document]
Comment (ZiyanShi) [reply include/kvdk/types.hpp:2]: `#pragma once` works poorly under some build systems, sometimes is ignored if relative path and absolute path coexist.
Traditional header guard is actually more robust.

File: include/kvdk/types.hpp (hunk 1 above)
Comment (ZiyanShi) [reply include/kvdk/types.hpp:21]: fixed

File: include/kvdk/types.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (sherry-1001) [reply include/kvdk/types.hpp:2]: so, align the other file

**Review by sherry-1001 — APPROVED** [review]

**Pull request merged**